pkg/beacon/relay/entry/states.go

```diff
@@ -125,6 +125,12 @@ func (scs *signatureCompleteState) Initiate(ctx context.Context) error {
 		scs.MemberIndex(),
 	)
 
+	previousEntryG1 := new(bn256.G1)
+	_, err := previousEntryG1.Unmarshal(scs.previousEntry)
+	if err != nil {
+		return err
+	}
```

Comment:
What if we do this in `entry.SignAndSubmit` and pass `bn256.G1` as `previousEntry` to `signingStateBase`? 

`previousEntry` is currently represented as `byte[]` and is used in two places:
- `sss.signer.CalculateSignatureShare` where it's unmarshalled to G1 before passing to `bls.SignG1`
- in this function where it is also unmarshalled to G1 before doing anything else.

Reply:
Good point! Thanks!